Dockerfile

```
@@ -0,0 +1,32 @@
+ARG GO_VERSION=1.22
+FROM golang:${GO_VERSION} AS builder
+
+ARG CADDY_VERSION=2.8.4
+RUN go install github.com/caddyserver/xcaddy/cmd/xcaddy@latest && \
+  xcaddy build v${CADDY_VERSION} --with github.com/tailscale/caddy-tailscale@main --output /usr/bin/caddy
```

```suggestion
RUN go run github.com/caddyserver/xcaddy/cmd/xcaddy@latest build v${CADDY_VERSION} \
--with github.com/tailscale/caddy-tailscale@main --output /usr/bin/caddy
```

maybe makes it a little faster?

---

Also, I'm wondering if we can use the `tailscale-caddy` repo checkout available to this action rather than refer to `github.com/tailscale/caddy-tailscale@main`? The checkout should presumably bet the one corresponding to the push event

---

I'd be surprised if `go run` was actually any faster, but certainly doesn't hurt so let's give it a try.

And oops, totally forgot to instruct it to use the local source for building.  I'll add that.